test/gtest/interpolate.hpp

```diff
@@ -0,0 +1,378 @@
+                  << " scale_factors:" << tc.scale_factors << " mode:" << tc.mode
+                  << " align_corners:" << tc.align_corners;
+    }
+
+    std::vector<size_t> GetInput() const { return input; }
+};
+
+inline std::vector<InterpolateTestCase> InterpolateTestConfigs()
+{
+    return {
+        {{16, 256, 1, 1, 1}, {32, 32, 32}, {32, 32, 32}, MIOPEN_INTERPOLATE_MODE_TRILINEAR, false},
+        {{16, 256, 1, 1, 1}, {32, 32, 32}, {0, 0, 0}, MIOPEN_INTERPOLATE_MODE_TRILINEAR, true},
+        {{16, 256, 1, 1, 1}, {32, 32, 32}, {0, 0, 0}, MIOPEN_INTERPOLATE_MODE_NEAREST, false},
+        {{16, 256, 1, 1}, {32, 32}, {0, 0}, MIOPEN_INTERPOLATE_MODE_NEAREST, false},
+        {{16, 256, 1, 1}, {32, 32}, {0, 0}, MIOPEN_INTERPOLATE_MODE_BILINEAR, false},
+        {{16, 256, 1, 1}, {32, 32}, {0, 0}, MIOPEN_INTERPOLATE_MODE_BILINEAR, true},
+        {{16, 256, 20, 20}, {40, 40}, {2, 2}, MIOPEN_INTERPOLATE_MODE_BICUBIC, false},
+        {{16, 256, 20, 20}, {40, 40}, {0, 0}, MIOPEN_INTERPOLATE_MODE_BICUBIC, true},
+        {{16, 256, 1}, {32}, {0}, MIOPEN_INTERPOLATE_MODE_NEAREST, false},
+        {{16, 256, 1}, {32}, {0}, MIOPEN_INTERPOLATE_MODE_LINEAR, false},
+        {{16, 256, 1}, {32}, {0}, MIOPEN_INTERPOLATE_MODE_LINEAR, true},
+    };
+}
+
+inline std::vector<size_t> GetStrides(std::vector<size_t> input, bool contiguous)
+{
+    if(!contiguous)
+        std::swap(input.front(), input.back());
+    std::vector<size_t> strides(input.size());
+    strides.back() = 1;
+    for(int i = input.size() - 2; i >= 0; --i)
+        strides[i] = strides[i + 1] * input[i + 1];
+    if(!contiguous)
+        std::swap(strides.front(), strides.back());
+    return strides;
+}
+
+// FORWARD TEST
+template <typename T = float>
+struct InterpolateTestFwd : public ::testing::TestWithParam<InterpolateTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle      = get_handle();
+        interpolate_config = GetParam();
+
+        auto in_dim   = interpolate_config.GetInput();
+        auto size     = interpolate_config.size;
+        mode          = interpolate_config.mode;
+        align_corners = interpolate_config.align_corners;
+
+        if(mode != MIOPEN_INTERPOLATE_MODE_NEAREST)
+        {
+            scale_factors = tensor<float>{size.size()};
+            for(int i = 0; i < size.size(); i++)
+                scale_factors[i] = interpolate_config.scale_factors[i];
+        }
+        else
+        {
+            scale_factors = tensor<float>{3};
+            for(int i = 0; i < size.size(); i++)
+                scale_factors[i] = interpolate_config.scale_factors[i];
+            for(int i = size.size(); i < 3; i++)
+                scale_factors[i] = 0;
+        }
+
+        auto out_dim = std::vector<size_t>({in_dim[0], in_dim[1]});
+        for(int i = 0; i < size.size(); i++)
+        {
+            if(scale_factors[i] != 0)
+                out_dim.push_back(ceil(static_cast<size_t>(in_dim[i + 2] * scale_factors[i])));
```

Comment:
Why do you need to call `ceil` after cast to `size_t`? seems redundant.

Reply:
removed cast size_t

Reply:
changed to static_cast<size_t>(ceil())